### ISSUE TITLE
Exception when trying to open file : DLL FreeImage

### ISSUE THREAD

**Issue by ThomasPerraudin** [LBognanni/ImageViewer#4]
```
Consultez la fin de ce message pour plus de détails sur l'appel du débogage
juste-à-temps (JIT) à la place de cette boîte de dialogue.

************** Texte de l'exception **************
System.DllNotFoundException: Impossible de charger la DLL 'FreeImage': Le module spécifié est introuvable. (Exception de HRESULT : 0x8007007E)
   à FreeImageAPI.FreeImage.GetFileTypeFromHandle(FreeImageIO& io, fi_handle handle, Int32 size)
   à FreeImageAPI.FreeImage.OpenMultiBitmapFromStream(Stream stream, FREE_IMAGE_FORMAT& format, FREE_IMAGE_LOAD_FLAGS flags)
   à FreeImageAPI.FreeImageBitmap.LoadFromStream(Stream stream, FREE_IMAGE_FORMAT format, FREE_IMAGE_LOAD_FLAGS flags)
   à ImageViewer.ImageLoader.LoadBitmap(String fileName)
   à ImageViewer.ImageLoader.LoadImage(String fileName)
   à System.Threading.Tasks.Task`1.InnerInvoke()
   à System.Threading.Tasks.Task.Execute()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   à ImageViewer.TwoStepImageCache.<LoadImageAsync>d__8.MoveNext()


************** Assemblys chargés **************
mscorlib
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4420.0 built by: NET48REL1LAST_C
    CodeBase : file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
ImageViewer
    Version de l'assembly : 1.0.5.0
    Version Win32 : 1.0.5
    CodeBase : file:///C:/Program%20Files/CodeMade%20Image%20Viewer/ImageViewer.exe
----------------------------------------
System.Windows.Forms
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4400.0 built by: NET48REL1LAST_C
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4360.0 built by: NET48REL1LAST_C
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4390.0 built by: NET48REL1LAST_C
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4190.0 built by: NET48REL1LAST_B
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4455.0 built by: NET48REL1LAST_C
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4084.0 built by: NET48REL1
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Accessibility
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4084.0 built by: NET48REL1
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------
FreeImage-dotnet-core
    Version de l'assembly : 4.3.6.0
    Version Win32 : 4.3.6.0
    CodeBase : file:///C:/Program%20Files/CodeMade%20Image%20Viewer/FreeImage-dotnet-core.DLL
----------------------------------------
Microsoft.WindowsAPICodePack.Shell
    Version de l'assembly : 1.1.0.0
    Version Win32 : 1.1.0.0
    CodeBase : file:///C:/Program%20Files/CodeMade%20Image%20Viewer/Microsoft.WindowsAPICodePack.Shell.DLL
----------------------------------------
WindowsBase
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4455.0 built by: NET48REL1LAST_C
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/WindowsBase/v4.0_4.0.0.0__31bf3856ad364e35/WindowsBase.dll
----------------------------------------
Microsoft.WindowsAPICodePack
    Version de l'assembly : 1.1.0.0
    Version Win32 : 1.1.0.0
    CodeBase : file:///C:/Program%20Files/CodeMade%20Image%20Viewer/Microsoft.WindowsAPICodePack.DLL
----------------------------------------
mscorlib.resources
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4084.0 built by: NET48REL1
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_fr_b77a5c561934e089/mscorlib.resources.dll
----------------------------------------
System.Windows.Forms.resources
    Version de l'assembly : 4.0.0.0
    Version Win32 : 4.8.4084.0 built by: NET48REL1
    CodeBase : file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_fr_b77a5c561934e089/System.Windows.Forms.resources.dll
----------------------------------------

************** Débogage JIT **************
Pour activer le débogage juste-à-temps (JIT), le fichier de configuration pour cette
application ou cet ordinateur (machine.config) doit avoir la valeur
jitDebugging définie dans la section system.windows.forms.
L'application doit également être compilée avec le débogage
activé.

Par exemple :

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

Lorsque le débogage juste-à-temps est activé, les exceptions non gérées
seront envoyées au débogueur JIT inscrit sur l'ordinateur
plutôt que d'être gérées par cette boîte de dialogue.



```

![image](https://user-images.githubusercontent.com/1092169/152107443-375105a6-b37c-4e21-8a23-8c92df182fe0.png)

Digitalis_AE.png :
![Digitalis_AE](https://user-images.githubusercontent.com/1092169/152107467-be38b15f-f7d3-4e7a-8edd-fd11c7c9f752.png)

System :
- Windows 10 Family Version 20H2 Build 19042.1466
- Windows Feature Experience Pack 120.2212.3920.0
- x64

**Comment by LBognanni** [comment]
Hey @ThomasPerraudin , could you try with the [lastest release](https://github.com/LBognanni/ImageViewer/releases/tag/1.3.0)?
I updated the build / installer workflow and now FreeImage should get bundled correctly. 
You will have to uninstall the previous version if you still have it 🙂

**Comment by ThomasPerraudin** [comment]
I still have the exact same issue. (I removed the previous version)

**Comment by LBognanni** [comment]
Hmmm if you navigate to `C:\Program Files\CodeMade Image Viewer\`, do you see the FreeImage.dll file? 
![image](https://user-images.githubusercontent.com/8505972/164904141-e7eac61f-e73e-4f00-8fc5-77019cee04ac.png)

**Comment by ThomasPerraudin** [comment]
Yes
![image](https://user-images.githubusercontent.com/1092169/164909181-3469ba0a-5ddd-4ec6-bb21-f9900d74d534.png)

**Comment by LBognanni** [comment]
Interesting, I wonder if it's somehow related to the french version of windows 🤔 
If you open ImageViewer.exe from there, does it still not work?

**Comment by ThomasPerraudin** [comment]
> If you open ImageViewer.exe from there, does it still not work?
I doesn't work.

**Comment by LBognanni** [comment]
After some reading, it appears to be because FreeImage requires the [Microsoft Visual C++ Runtime 2015](https://www.microsoft.com/en-gb/download/details.aspx?id=48145) to be installed.
Could you try to install it and report back? 
If that's the case I'll look into bundling it with the regular setup 🤔

**Comment by LBognanni** [comment]
I was able to reproduce this on a fresh VM. 
The missing package is actually ["Visual C++ Redistributable Packages for Visual Studio 2013"](https://www.microsoft.com/en-us/download/details.aspx?id=40784) 🤦‍♂️

**Comment by LBognanni** [comment]
Closing as this is now fixed in v 1.4.0, which includes the VC++ package in the setup. 
Feel free to reopen or open a new issue in case it still doesn't work for you 🙂

**Comment by ThomasPerraudin** [comment]
- It works !
- Be careful with the installation process : Windows appears to prevent installation: (I had to click on "Informations complémentaires" to install the app)
![image](https://user-images.githubusercontent.com/1092169/164972712-e7af8836-c0d4-4ae6-b0f4-9110893e1280.png)
- At the end of the installation, the checkbox "Run the app" was checked but the app never started.
